### PR TITLE
AST NFC: Minor cleanups around ASTContext

### PR DESCRIPTION
**Pull request by Azoy**
This is part 1 for a series of cleanups I want to do around known items in the compiler. This patch specifically just moves Void into a known stdlib type, and `_unimplementedInitializer` and `_undefined` into known functions.

cc: @slavapestov @jrose-apple

### CODE REVIEW

**Review by slavapestov — APPROVED** [review]

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Review by jrose-apple — APPROVED** [review]

**Comment by Azoy** [comment]
@slavapestov alright this should be ready to go.

File: lib/AST/ASTContext.cpp [hunk 1]
```diff
@@ -696,6 +696,7 @@ FuncDecl *ASTContext::getPlusFunctionOnString() const {
       if (auto type = dyn_cast<DECL_CLASS>(result)) { \
         auto params = type->getGenericParams(); \
         if (NUM_GENERIC_PARAMS == (params == nullptr ? 0 : params->size())) { \
+          getImpl().NAME##Decl = type; \
```
Comment (jrose-apple) [inline lib/AST/ASTContext.cpp]: !

Comment (Azoy) [reply lib/AST/ASTContext.cpp]: ?

Comment (jrose-apple) [reply lib/AST/ASTContext.cpp]: This change implies that this has never been cached, ever, right? (Or at least since the last refactoring.

Comment (Azoy) [reply lib/AST/ASTContext.cpp]: It was being cached, but the control flow in the previous version made it a little unintuitive.

Comment (jrose-apple) [reply lib/AST/ASTContext.cpp]: Oh, whoops, I saw the partial diff from your previous version and thought it was a complete change. This makes sense.

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
⛵️

**Comment by Azoy** [comment]
Thanks @CodaFi !

**Pull request merged**